Jellyfin.Server.Implementations/Users/DefaultPasswordResetProvider.cs

```diff
@@ -52,90 +63,96 @@ public class DefaultPasswordResetProvider : IPasswordResetProvider
         public bool IsEnabled => true;
 
         /// <inheritdoc />
-        public async Task<PinRedeemResult> RedeemPasswordResetPin(string pin)
+        public async Task<CodeRedeemResult> RedeemPasswordResetPin(string code, string password)
         {
-            var usersReset = new List<string>();
-            foreach (var resetFile in Directory.EnumerateFiles(_passwordResetFileBaseDir, $"{BaseResetFileName}*"))
+            SerializablePasswordReset passwordReset;
+            foreach (var file in Directory.EnumerateFiles(_passwordResetFileBaseDir, $"{BaseResetFileName}*"))
             {
-                SerializablePasswordReset spr;
-                await using (var str = File.OpenRead(resetFile))
+                using (var stream = File.OpenRead(file))
                 {
-                    spr = await _jsonSerializer.DeserializeFromStreamAsync<SerializablePasswordReset>(str).ConfigureAwait(false);
+                    passwordReset = await _jsonSerializer.DeserializeFromStreamAsync<SerializablePasswordReset>(stream).ConfigureAwait(false);
                 }
 
-                if (spr.ExpirationDate < DateTime.UtcNow)
+                if (passwordReset.ExpirationDate < DateTime.Now)
                 {
-                    File.Delete(resetFile);
+                    File.Delete(file);
                 }
-                else if (string.Equals(
-                    spr.Pin.Replace("-", string.Empty, StringComparison.Ordinal),
-                    pin.Replace("-", string.Empty, StringComparison.Ordinal),
-                    StringComparison.InvariantCultureIgnoreCase))
+                else if (string.Equals(passwordReset.Code, code, StringComparison.InvariantCultureIgnoreCase))
                 {
-                    var resetUser = _userManager.GetUserByName(spr.UserName)
-                        ?? throw new ResourceNotFoundException($"User with a username of {spr.UserName} not found");
-
-                    await _userManager.ChangePassword(resetUser, pin).ConfigureAwait(false);
-                    usersReset.Add(resetUser.Username);
-                    File.Delete(resetFile);
+                    var resetUser = _userManager.GetUserByName(passwordReset.Username);
+                    if (resetUser == null)
+                    {
+                        File.Delete(file);
+                        continue;
+                    }
+
+                    await _userManager.ChangePassword(resetUser, password).ConfigureAwait(false);
+                    File.Delete(file);
```

Comment:
No exception handling is present. I would add a try catch for logging at least should the file not be deleted.